src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DataFlowUtilities.cs

```diff
@@ -205,6 +205,92 @@ public static IDisposable LinkToAsyncAction<T>(this ISourceBlock<T> source, Func
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="source"/> is <see langword="null"/>.
+        ///     <para>
+        ///         -or-
+        ///     </para>
+        ///     <paramref name="transform"/> is <see langword="null"/>.
+        /// </exception>
+        public static DisposableValue<ISourceBlock<TOut>> TransformWithNoDelta<TInput, TOut>(this ISourceBlock<IProjectVersionedValue<TInput>> source, Func<IProjectVersionedValue<TInput>, TOut> transform)
+        {
+            Requires.NotNull(source, nameof(source));
+            Requires.NotNull(transform, nameof(transform));
+
+            IPropagatorBlock<IProjectVersionedValue<TInput>, TOut> transformBlock = DataflowBlockSlim.CreateTransformBlock(transform, skipIntermediateInputData: true, skipIntermediateOutputData: true);
+
+            IDisposable link = source.LinkTo(transformBlock, DataflowOption.PropagateCompletion);
+
+            return new DisposableValue<ISourceBlock<TOut>>(transformBlock, link);
+        }
+
+        /// <summary>
+        ///     Creates a source block that produces a transformed value for each value from original source block, 
+        ///     skipping intermediate input and output states, and hence is not suitable for producing or consuming 
+        ///     deltas.
+        /// </summary>
+        /// <typeparam name="TOut">
+        ///     The type of value produced by <paramref name="transform"/>.
+        ///  </typeparam>
+        /// <param name="source">
+        ///     The source block whose values are to be transformed.
+        /// </param>
+        /// <param name="transform">
+        ///     The function to execute on each value from <paramref name="source"/>.
+        /// </param>
+        /// <param name="suppressVersionOnlyUpdates">
+        ///     A value indicating whether to prevent messages from propagating to the target
+        ///     block if no project changes are include other than an incremented version number.
+        /// </param>
+        /// <param name="ruleNames">
+        ///     The names of the rules that describe the project data the caller is interested in.
+        /// </param>
+        /// <returns>
+        ///     The transformed source block and a disposable value that terminates the link.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="source"/> is <see langword="null"/>.
+        ///     <para>
+        ///         -or-
+        ///     </para>
+        ///     <paramref name="transform"/> is <see langword="null"/>.
+        /// </exception>
+        public static DisposableValue<ISourceBlock<TOut>> TransformWithNoDelta<TOut>(this ISourceBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>> source, Func<IProjectVersionedValue<IProjectSubscriptionUpdate>, TOut> transform, bool suppressVersionOnlyUpdates = true, IEnumerable<string> ruleNames = null)
```

Comment:
it is easy to make a mistake to turn on "suppressVersionOnlyUpdates" to cause dead locks and other problems (with operation progress feature).  My suggestion is that we don't provide the default value (which is true), and force the developer to think about it and make the right decision whenever they are using it.  It also makes it easier to code review (so you will review the place where it is used to see whether this option is set correctly.)

Reply:
Fixed this - we are used to be ending of the chain so my initial feeling was that this should be true by default, but this is a transform - and hence likely in the middle of a chain. Agree and fixed this.